test/syncenginetestutils.h

```diff
@@ -649,6 +649,223 @@ class FakeErrorReply : public QNetworkReply
     qint64 readData(char *, qint64) override { return 0; }
 };
 
+class FakeBundlePOSTReply : public QNetworkReply
+{
+    Q_OBJECT
+    FileInfo *fileInfo;
+    QByteArray payload;
+public:
+    FakeBundlePOSTReply(FileInfo &remoteRootFileInfo, QNetworkAccessManager::Operation op, const QNetworkRequest &request, const QByteArray &postPayload, QObject *parent)
+    : QNetworkReply{parent} {
+        setRequest(request);
+        QUrl rawUrl = request.url();
+        QString bundlePath(sBundleRootUrl.path()+rawUrl.userName());
+        setUrl(rawUrl);
+        setOperation(op);
+        open(QIODevice::ReadOnly);
+        const QString davUri{QStringLiteral("DAV:")};
+        const QString ocUri{QStringLiteral("http://owncloud.org/ns")};
+        const QString sabUri{QStringLiteral("http://sabredav.org/ns")};
+        QBuffer buffer{&payload};
+        buffer.open(QIODevice::WriteOnly);
+        QXmlStreamWriter xml( &buffer );
+        xml.writeNamespace(davUri, "d");
+        xml.writeNamespace(ocUri, "o");
+        xml.writeNamespace(sabUri, "s");
+
+        if ("erroruser" == rawUrl.userName()) {
+            // ERROR CASE
+            xml.writeStartDocument();
+            xml.writeStartElement(davUri, QStringLiteral("error"));
+            xml.writeTextElement(sabUri, QStringLiteral("exception"), QStringLiteral("OCA\\DAV\\Connector\\Sabre\\Exception\\Forbidden"));
+            xml.writeTextElement(sabUri, QStringLiteral("message"), QStringLiteral("URL endpoint has to be instance of \\OCA\\DAV\\Files\\FilesHome"));
+            xml.writeTextElement(ocUri, QStringLiteral("retry"), QStringLiteral("false"));
+            xml.writeTextElement(ocUri, QStringLiteral("reason"), QStringLiteral("URL endpoint has to be instance of \\OCA\\DAV\\Files\\FilesHome"));
+            xml.writeEndElement(); // error
+            xml.writeEndDocument();
+            setAttribute(QNetworkRequest::HttpStatusCodeAttribute, 403);
```

Comment:
I am trying to generate test, which will throw error for whole the bundle. This does not work. @jturcotte